utils/swift_build_support/swift_build_support/products/minimalstdlib.py

```diff
@@ -0,0 +1,146 @@
+#
+# ----------------------------------------------------------------------------
+
+import os
+import platform
+
+from . import cmake_product
+from . import cmark
+from . import libcxx
+from . import llvm
+from . import swift
+
+
+class MinimalStdlib(cmake_product.CMakeProduct):
+    @classmethod
+    def product_source_name(cls):
+        return os.path.join("swift")
+
+    @classmethod
+    def is_build_script_impl_product(cls):
+        return False
+
+    @classmethod
+    def is_before_build_script_impl_product(cls):
+        return False
+
+    def should_build(self, host_target):
+        return platform.system() == 'Darwin'
+
+    def build(self, host_target):
+        build_variant = 'MinSizeRel'
+        self.cmake_options.define('CMAKE_BUILD_TYPE:STRING', build_variant)
+        self.cmake_options.define('SWIFT_STDLIB_BUILD_TYPE:STRING', build_variant)
+
+        build_root = os.path.dirname(self.build_dir)
+        llvm_build_dir = os.path.join('..', build_root, '%s-%s' % ('llvm', host_target))
+        llvm_cmake_dir = os.path.join(llvm_build_dir, 'lib', 'cmake', 'llvm')
+        self.cmake_options.define('LLVM_DIR:PATH', llvm_cmake_dir)
+
+        # Use the just-built toolchain
+        toolchain_dir = self.install_toolchain_path(host_target)
+        self.cmake_options.define('TOOLCHAIN_DIR:PATH', toolchain_dir)
+        self.cmake_options.define('SWIFT_NATIVE_CLANG_TOOLS_PATH:STRING', toolchain_dir + '/bin')
+        self.cmake_options.define('SWIFT_NATIVE_LLVM_TOOLS_PATH:STRING', toolchain_dir + '/bin')
+        self.cmake_options.define('SWIFT_NATIVE_SWIFT_TOOLS_PATH:STRING', toolchain_dir + '/bin')
+
+        # Build the freestanding variant, with these options
+        self.cmake_options.define('SWIFT_SDKS:STRING', 'FREESTANDING')
+        self.cmake_options.define('SWIFT_FREESTANDING_ARCHS:STRING', 'x86_64;arm64')
+        self.cmake_options.define('SWIFT_FREESTANDING_FLAVOR:STRING', 'apple')
+        self.cmake_options.define('SWIFT_FREESTANDING_IS_DARWIN:BOOL', 'FALSE')
+        self.cmake_options.define('SWIFT_FREESTANDING_MODULE_NAME:STRING', 'macos')
+        self.cmake_options.define('SWIFT_FREESTANDING_SDK:STRING', 'macosx')
+        self.cmake_options.define('SWIFT_FREESTANDING_TRIPLE_NAME:STRING', 'macosx11.0')
+        self.cmake_options.define('SWIFT_PRIMARY_VARIANT_ARCH:STRING', 'x86_64')
+        self.cmake_options.define('SWIFT_PRIMARY_VARIANT_SDK:STRING', 'FREESTANDING')
+        self.cmake_options.define('SWIFT_HOST_TRIPLE:STRING', 'x86_64-apple-macosx10.13')
+        self.cmake_options.define('SWIFT_HOST_VARIANT', 'macosx')
+        self.cmake_options.define('SWIFT_HOST_VARIANT_ARCH', 'x86_64')
+        self.cmake_options.define('SWIFT_HOST_VARIANT_SDK', 'OSX')
+        self.cmake_options.define('SWIFT_STDLIB_ASSERTIONS:BOOL', 'FALSE')
+
+        # Configure build to only build the stdlib, and only a static version of it
+        self.cmake_options.define('SWIFT_BUILD_DYNAMIC_SDK_OVERLAY:BOOL', 'FALSE')
```

Comment:
Note to self: This needs to somehow be unified with the settings in build-preset.ini